docs/site/Req-res-cycle.md

```diff
@@ -0,0 +1,278 @@
+
+{% include tip.html content="Use controller methods for creating custom
+endpoints instead of mounting Express routers. Controller methods support
+dependency injection and have access to the whole application context, while
+a Express routes don't." %}
+
+##### Services
+
+[Services](https://loopback.io/doc/en/lb4/Services.html) are LoopBack's helper
+units with support for dependency injection. They come in three varieties -
+proxy, class, and provider.
+
+A proxy service acts as a proxy to an external service. A class service creates
+an instance of a helper class. A provider service resolves a value.
+
+All of them have access to the app, request, and response objects via dependency
+injections, and can influence the handling of the request-response cycle.
+
+Services can be injected in the controller constructor so controller methods may
+use them.
+
+##### Repositories
+
+[Repositories](https://loopback.io/doc/en/lb4/Repositories.html) are the links
+between controllers and the data. They use an underlying datasource and
+a connector to interact with the data.
+
+Controller methods can call corresponding methods in the respository to read
+from or write to the database.
+
+Repositories can be injected in the controller constructor so controller methods
+may use them.
+
+##### Interceptors
+
+[Interceptors](https://loopback.io/doc/en/lb4/Interceptors.html) intercept
+invocation of method on classes decorate with the `@intercept()` decorator.
+
+All controllers, services, and repositories; and their methods can be intercepted.
+This makes interceptors a powerful participant in the request-response cycle,
+since they can modify the request and response objects and call their methods.
+
+{% include tip.html content="The [authorization component](https://loopback.io/doc/en/lb4/apidocs.authorization.html)
+is an example of the use of interceptors to add additional functionality to the
+app." %}
+
+##### Walk through of a request
+
+Now that we know all the components that may be involve in a request to a
+controller endpoint, let's walk through a request to an endpoint - `/ping`
+endpoint of a scaffolded LoopBack app.
+
+The `PingController` controller has a method named `ping()`, decorated with a
+`get()` decorator. This setup creates the `/ping` endpoint on the app.
+
+The app is configured to use the `MySequence` sequence.
+
+Now this is a step-wise sequence of what happens when a request is made to
+`http://localhost:4000/ping`.
+
+1. The request is intercepted by `MySequence`.
+2. The route is then identified by `FindRoute`. There is a handler for this request.
+3. `ParseParams` then tries to parses any parameters might have been submitted.
+There is none.
+4. `InvokeMethod` then invokes the handler with empty parameters. It returns
+whatever the handler returns, in this case, it is JSON object with `greeting`,
+`date`, `url`, and `headers` properties.
+5. `Send` then sends this JSON object back to the client.
```

Comment:
Let's mention error handling here for `Reject`.

Reply:
Added: `6. `Reject` is in place to catch any errors that might be thrown in any
of the steps above.`